### PR TITLE
Add doc to dataloader

### PR DESCRIPTION
**Pull request by tonywu71**
@alexandrashaw10 @William-Baker This PR is mostly about adding comments to the existing code in `main`.

However, I did change the name of the `RegressionDataGeneratorUniform` class to `RegressionDataGeneratorArbitraryGP` because technically speaking only the x values are sampled via a uniform distribution. Hence the possible confusion.

### CODE REVIEW

File: dataloader/load_regression_data_from_arbitrary_gp.py [hunk 1]
```diff
@@ -72,38 +85,51 @@ def gen(batch_size,
...
+        assert min_val_uniform < max_val_uniform, "min_val_uniform must be smaller than max_val_uniform"
+        self.min_val_uniform = min_val_uniform
+        self.max_val_uniform = max_val_uniform
+        
         self.kernel_length_scale = kernel_length_scale
         
         self.train_ds, self.test_ds = self.load_regression_data()
 
     
     def get_gp_curve_generator_from_uniform(self, testing: bool=False) -> Callable:
```
Comment (alexandrashaw10) [inline dataloader/load_regression_data_from_arbitrary_gp.py:116]: should this name be changed as well?

Comment (tonywu71) [reply dataloader/load_regression_data_from_arbitrary_gp.py:116]: Or good catch! I'll do it right away.

Comment (tonywu71) [reply dataloader/load_regression_data_from_arbitrary_gp.py:116]: Good catch! I'll change them to `min_x_val_uniform` and `max_x_val_uniform` to make it more clear.

**Comment by tonywu71** [comment]
@alexandrashaw10 @William-Baker If you do encounter some merge conflicts due to this PR, send me a message and I'll help you with solving the conflicts.

**Pull request merged**